cumulusci/tasks/salesforce/content_documents.py

```diff
@@ -0,0 +1,237 @@
+                    record["Id"] for record in self.sf.query_all(query)["records"]
+                ]
+                if record_ids:
+                    self.logger.info(
+                        f"        ({len(record_ids)}) {', '.join(record_ids)}"
+                    )
+                    all_record_ids.update(record_ids)
+                else:
+                    self.logger.info("        🚫 No records found.")
+        else:
+            self.logger.info("    No queries specified.")
+        return list(all_record_ids)
+
+    def _link_records_to_content_document(
+        self, content_document_id: str, record_ids: list[str]
+    ):
+        self.logger.info("")
+        if record_ids:
+            share_type = self.options["share_type"]
+            visibility = self.options["visibility"]
+            self.logger.info(
+                "Inserting ContentDocumentLink records to link the ContentDocument."
+            )
+            self.logger.info(f'    ShareType: "{share_type}"')
+            self.logger.info(f'    Visibility: "{visibility}"')
+            for record_id in record_ids:
+                self.sf.ContentDocumentLink.create(
+                    {
+                        "ContentDocumentId": content_document_id,
+                        "LinkedEntityId": record_id,
+                        "ShareType": share_type,
+                        "Visibility": visibility,
+                    }
+                )
+            self.logger.info(
+                f'Successfully linked {len(record_ids)} record{"" if len(record_ids) == 1 else "s"} to Content Document "{content_document_id}"'
+            )
+        else:
+            self.logger.info(
+                "😴 No records IDs queried. Skipping linking the Content Document to related records."
+            )
+
+    def _run_task(self):
+        # Insert the ContentDocument.
+        content_document_id = self._insert_content_document()
+
+        # Query records to link to the new ContentDocument.
+        # "Rolls back" the ContentDocument insert if an Exception is raised.
+        record_ids = None
+        get_record_ids_to_link_exception = None
+        try:
+            record_ids = self._get_record_ids_to_link()
+        except SalesforceMalformedRequest as e:
+            # Convert SalesforceMalformedRequest into something easier to read
+            get_record_ids_to_link_exception = to_cumulusci_exception(e)
+        except Exception as e:
+            get_record_ids_to_link_exception = e
+        finally:
+            if get_record_ids_to_link_exception:
+                self.logger.error(
+                    "An error occurred querying records to link to the ContentDocument."
+                )
+                # Reraise the Exception
+                raise get_record_ids_to_link_exception
+
+        # Links queried records to link to the new ContentDocument.
+        # "Rolls back" the ContentDocument insert if an Exception is raised.
```

Comment:
We are not currently performing a rollback, correct?

Reply:
Correct.  We removed that feature, but I forgot to update the comments.  Thank you!